### DISCUSSION TITLE
HomeMaticDimmerAccessory SubTypes wie bei HomeMaticSwitchAccessory?

### DISCUSSION THREAD

**Discussion by jb-home** [thkl/hap-homematic#339] [Q&A]
Hallo @thkl,

ist es in HomeKit möglich, einem HomeMaticDimmerAccessory die gleichen sub Typen zu verpassen wie dem  HomeMaticSwitchAccessory? Also die Wahl zwischen 'Lightbulb', 'Outlet', 'Switch', 'Fan' ?

Da ich einen HomeMatic Dimmer Stecker benutze, um einen Lüfter zu steuern. Wäre halt super, dass wenn man alle Lichter aus macht, der Lüfter dann nicht ausgehen würde z.B.

**Comment by jb-home** [comment]
@thkl ist das technisch überhaupt möglich?

**Reply by thkl** [reply]
Technisch geht das schon.

Denn Fan kennt die Eigenschaft Characteristic.RotationSpeed

**Reply by thkl** [reply]
Ok als Extra Typ beim Dimmer geht es nicht. Aber nur weil die DImmerklasse eine Superklasse für RGB Geräte ist.
Dann gäbe es auch in den RGB Geräten die Möglichkeit einen Fan einzustellen, was quatsch ist.

Deswegen habe ich einfach eine eigene Fan Klasse gebaut, die auf quasi den Dimmer kopiert und nur auf den Fön anpasst.

**Reply by jb-home** [reply]
OK, kommt dann mit dem nächsten release?

**Reply by jb-home** [reply]
Ich kann dann also z.B. einem HM-LC-Dim1T-Pl-3 statt der Dimmer Klasse die Fan Klasse zuweisen und in HomeKit wird der dann automatisch als Fan/Lüfter dargestellt?

**Reply by thkl** [reply]
Yes, Allen DIMMER oder VIRTUAL_DIMMER Kanälen.  Und die bekommen dann sogar eine Drehzahlsteuerung via LEVEL Datenpunkt.
